File: 2015/01/solution.rb
# frozen_string_literal: true

input = $stdin.read.chars

input = input.map { |c| c == ')' ? -1 : 1 }
accum = input.each_with_object([0]) { |a, acc| acc << acc[-1] + a }.to_a

part1 = accum[-1]
part2 = accum.find_index(-1) + 1

p [part1, part2]

File: 2015/01/solution.clj
(def input
  (->>
   (slurp "in.txt")
   (map #(if (= % \() 1 -1))))

(let
    [reds (reductions + input)
     part1 (last reds)
     part2 (.indexOf reds -1)]
  (println part1 part2))
